### PR TITLE
fix(style): form  field error border style for select

### PR DESCRIPTION
**Pull request by jyash97**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#20482 
### 💡 Background and solution

Another border was getting added due to some internal style of select. I had to override it to fix the issue.

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/form/style/index.less [hunk 1]
```diff
@@ -295,6 +295,14 @@
       }
     }
 
+    // fixes https://github.com/ant-design/ant-design/issues/20482
+    .@{ant-prefix}-input-group-addon .@{ant-prefix}-select {
+      &.@{ant-prefix}-select-single:not(.@{ant-prefix}-select-customize-input)
+        .@{ant-prefix}-select-selector {
+        border: 0;
+      }
+    }
```
Comment (zombieJ) [inline components/form/style/index.less]: Remove this line: https://github.com/ant-design/ant-design/blob/7998a3312e780cec232f6b1d5ad389c6f9a63288/components/form/style/index.less#L305

Comment (jyash97) [reply components/form/style/index.less]: Hey @zombieJ , sorry I didn't understand.  Did you mean to remove the `&-selector` on `L305` and add the class name for `selector` like below ?
```
.@{ant-prefix}-input-group-addon .@{ant-prefix}-select .@{ant-prefix}-select-selector {
        border-color: transparent;
        box-shadow: none;
  }
```

Comment (zombieJ) [reply components/form/style/index.less]: I mean `L305` is a fix for this in v3 but not work in v4. If you add new patch to fix this, old one should be removed.

Comment (jyash97) [reply components/form/style/index.less]: Cool thanks 👍 Have pushed the fix

**Comment by jyash97** [comment]
Thanks, @zombieJ for the quick reviews on my PR.

**Pull request merged**